### PR TITLE
ci: do static checks using golangci-lint

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Changelog: None
Ticket: QA-235

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
-  - project: 'Northern.tech/Mender/mendertesting'
-    file: '.gitlab-ci-check-license.yml'
-  - project: 'Northern.tech/Mender/mendertesting'
-    file: '.gitlab-ci-github-status-updates.yml'
+    file:
+      - '.gitlab-ci-check-golang-lint.yml'
+      - '.gitlab-ci-check-golang-unittests-v2.yml'
+      - '.gitlab-ci-check-commits-signoffs.yml'
+      - '.gitlab-ci-check-license.yml'
+      - '.gitlab-ci-github-status-updates.yml'
```
Comment (merlin-northern) [inline .gitlab-ci.yml:8]: so they are not from the mendertesting project?

Comment (kjaskiewiczz) [reply .gitlab-ci.yml:8]: they are, but you can simply list all the files for the project instead of adding project, file pairs

**Review by tranchitella — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (merlin-northern) [reply .gitlab-ci.yml:8]: ah ok you left the first project line there. sorry I missed that.

**Review by merlin-northern — APPROVED** [review]

**Review by merlin-northern — APPROVED** [review]
looks good left some annoying comments. great work.

File: mongo/dbtest/db.go [hunk 2]
```diff
@@ -127,7 +127,9 @@ func (dbs *DBServer) start() {
 }
 
 func (dbs *DBServer) monitor() error {
-	dbs.server.Process.Wait()
+	if _, err := dbs.server.Process.Wait(); err != nil {
+		fmt.Fprintf(os.Stderr, "---- error waiting for the process to exit: %s\n", err)
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply mongo/dbtest/db.go:131]: yes

File: mongo/dbtest/db.go [hunk 3]
```diff
@@ -136,7 +138,9 @@ func (dbs *DBServer) monitor() error {
 		cmd := exec.Command("/bin/sh", "-c", "ps auxw | grep mongod")
 		cmd.Stdout = os.Stderr
 		cmd.Stderr = os.Stderr
-		cmd.Run()
+		if err := cmd.Run(); err != nil {
+			fmt.Fprintf(os.Stderr, "---- error executing command: %s\n", err)
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply mongo/dbtest/db.go:142]: yes

File: mongo/dbtest/db.go [hunk 4]
```diff
@@ -191,7 +201,8 @@ func (dbs *DBServer) Client() *mongo.Client {
 			dbs.timeout = 8
 		}
 		clientOptions := options.Client().ApplyURI("mongodb://" + dbs.host + "/test")
-		dbs.Ctx = context.Background() // context.WithTimeout(context.Background(), dbs.timeout*time.Second)
+		// context.WithTimeout(context.Background(), dbs.timeout*time.Second)
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply mongo/dbtest/db.go:204]: I think the comment is actually from you ;), but I can remove it

File: mongo/migrate/migrator_dummy.go [hunk 5]
```diff
@@ -51,7 +51,9 @@ func (m *DummyMigrator) Apply(ctx context.Context, target Version, migrations []
 
 	if !m.Automigrate {
 		if VersionIsLess(last, target) {
-			return fmt.Errorf(ErrNeedsMigration+": %s has version %s, needs version %s", m.Db, last.String(), target.String())
+			return fmt.Errorf(
+				ErrNeedsMigration+": %s has version %s, needs version %s",
+				m.Db, last.String(), target.String())
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply mongo/migrate/migrator_dummy.go]: done

File: mongo/migrate/migrator_simple.go [hunk 6]
```diff
@@ -83,7 +85,9 @@ func (m *SimpleMigrator) Apply(ctx context.Context, target Version, migrations [
 	// if the last applied migration is lower than the target one
 	if !m.Automigrate {
 		if VersionIsLess(last, target) {
-			return fmt.Errorf(ErrNeedsMigration+": %s has version %s, needs version %s", m.Db, last.String(), target.String())
+			return fmt.Errorf(
+				ErrNeedsMigration+": %s has version %s, needs version %s",
+				m.Db, last.String(), target.String())
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply mongo/migrate/migrator_simple.go]: done

File: requestid/middleware.go [hunk 7]
```diff
@@ -71,7 +71,8 @@ func Middleware(opts ...*MiddlewareOptions) gin.HandlerFunc {
 	}
 }
 
-// RequestIdMiddleware sets the X-MEN-RequestID header if it's not present, and and adds the request id to the request's logger's context.
+// RequestIdMiddleware sets the X-MEN-RequestID header if it's not present,
+// and and adds the request id to the request's logger's context.
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply requestid/middleware.go]: done

File: rest_utils/paging.go [hunk 8]
```diff
@@ -127,7 +137,8 @@ func ParsePagination(r *rest.Request) (uint64, uint64, error) {
 		return 0, 0, err
 	}
 
-	per_page, err := ParseQueryParmUInt(r, PerPageName, false, PerPageMin, PerPageMax, PerPageDefault)
+	per_page, err := ParseQueryParmUInt(
+		r, PerPageName, false, PerPageMin, PerPageMax, PerPageDefault)
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply rest_utils/paging.go]: done

File: routing/routing.go [hunk 9]
```diff
@@ -45,7 +45,7 @@ func AutogenOptionsRoutes(routes []*rest.Route, gen HttpOptionsGenerator) []*res
 	for _, route := range routes {
 		methods, ok := methodGroups[route.PathExp]
 		if !ok {
-			methods = make([]string, 0, 0)
+			methods = make([]string, 0)
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply routing/routing.go:48]: don't think so

File: accesslog/middleware.go [hunk 10]
```diff
@@ -240,7 +242,8 @@ func (mw *AccessLogMiddleware) convertFormat() {
 	}
 }
 
-// accessLogUtil provides a collection of utility functions that devrive data from the Request object.
+// accessLogUtil provides a collection of utility functions
+// that devrive data from the Request object.
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply accesslog/middleware.go]: done

**Pull request merged**